src/microbe_stage/MicrobeAI.cs

```diff
@@ -395,6 +398,60 @@ private void EngagePrey(Vector3 target, Random random, bool engulf)
         }
     }
 
+    private void SeekCompounds(Random random, MicrobeAICommonData data)
+    {
+        // If we are still engulfing for some reason, stop
+        microbe.State = Microbe.MicrobeState.Normal;
+
+        // More active species just try to get distance to avoid over-clustering
+        if (RollCheck(SpeciesActivity, Constants.MAX_SPECIES_ACTIVITY + (Constants.MAX_SPECIES_ACTIVITY / 2), random))
+        {
+            SetMoveSpeed(Constants.AI_BASE_MOVEMENT);
+            return;
+        }
+
+        if (random.Next(Constants.AI_STEPS_PER_SMELL) == 0)
+        {
+            SmellForComounds(data);
+        }
+
+        if (lastSmelledCompoundPosition != null)
+        {
+            if (DistanceFromMe(lastSmelledCompoundPosition.Value) > 3.0f)
+            {
+                targetPosition = lastSmelledCompoundPosition.Value;
+                microbe.LookAtPoint = targetPosition;
+            }
+            else
+            {
+                SmellForComounds(data);
+            }
+        }
+        else
+        {
+            RunAndTumble(random);
+        }
+    }
+
+    private void SmellForComounds(MicrobeAICommonData data)
+    {
+        ComputeCompoundsSearchWeights();
+
+        var detections = microbe.GetDetectedCompounds(data.Clouds)
```

Comment:
Shouldn't the AI have some kind of knowledge of when it *doesn't* have a chemoreceptor to not attempt to use it?

Reply:
I would argue that this is better separation of concerns. The Microbe class, which owns the detections, is capable of answering the question, "what do you smell?" by running a loop zero times and returning a list of zero items. Adding a check here would just be an extra place to break that doesn't protect against anything. You will notice that, so far, the AI makes no checks about the microbe's organelles.

Reply:
*cough* *cough* I just finished the signaling agents, and they *do* check a property on the Microbe to see if it has signaling agents. I think such a check is similar to checking if the memberane set on the Microbe allows engulfing or not, which is directly checked in the AI. I think that's less clear separation of concerns, a cleaner way would be for the AI to ask the Microbe class if it can engulf instead of reading the membrane type and comparing that.

Reply:
I agree, and maybe that would be a good future refactor. This code asks the Microbe class if it has any active detections instead of reading the organelle list and comparing that. Detections are currently only created when there are chemoreceptors and they smell something. If later on that changes, the change won't break this code.